packages/ui/src/mocks/providers/index.tsx

```diff
@@ -31,7 +33,9 @@ export const MockProvidersDecorator = (Story: CallableFunction, storyContext: Co
       <MockGqlProvider queries={mocks?.gql?.queries} mutations={mocks?.gql?.mutations}>
         <MockAccountsProvider accounts={mocks?.accounts}>
           <MockBackendProvider backend={mocks?.backend}>
-            <Story />
+            <MockRouteQueryProvider routeQuery={mocks?.routeQuery}>
```

Comment:
I created a PR where the router `href` is mocked directly: https://github.com/kdembler/pioneer/pull/1/files

I really prefer mocking on the router level rather than the `useRouteQuery` hook because not all component use this hook so it would make different tests mock urls differently.

Currently the router mock is a decorator added to `packages/ui/.storybook/preview.tsx` but if you prefer to add it here on `packages/ui/src/mocks/providers/index.tsx` to be able to change it's argument depending on e.g `args.isConfirmingEmail` like you did with this `MockRouteQueryProvider` that's even better!